pagopa/io-app
# chore: [IOAPPFD0-75] Add public key thumbprint in Profile debug info

## Short description
This PR adds the current public key thumbprint in the `ProfileMainScreen` if debug mode is enabled.

## List of changes proposed in this pull request
- [ts/features/lollipop/utils/crypto.ts](https://github.com/pagopa/io-app/pull/4504/files#diff-e70e8137cd500fa6551b90da08d74031c1c74ef09f7b414cdbec642e72acbf4e): adds two crypto utility functions to generate the public key thumbprint.
- [ts/features/lollipop/utils/__tests__/thumbprint.test.ts](https://github.com/pagopa/io-app/pull/4504/files#diff-c62d0c1b763402b50132882e178a733c55d5dc599fb1cc6f4d41bf3680decd9d): adds test to check the above utility functions.
- [ts/features/lollipop/utils/login.ts](https://github.com/pagopa/io-app/pull/4504/files#diff-f54ddab5813a42b9135c11cbd42b931854b83272a0763ec2fd2b759f8ede6f09): makes use of the new functions for thumbprint.
- [ts/features/lollipop/utils/login.ts](https://github.com/pagopa/io-app/pull/4504/files#diff-f54ddab5813a42b9135c11cbd42b931854b83272a0763ec2fd2b759f8ede6f09): as above.
- [ts/screens/profile/ProfileMainScreen.tsx](https://github.com/pagopa/io-app/pull/4504/files#diff-de3c1a64bcc8fdca4c8dbc5d62dba36c8c5ded27255052ecc6526e614b1e8f5b): adds the thumbprint list item in debug mode.

## How to test
Go to `ProfileMainScreen`, then enable debug mode. Scroll down until you see the thumbprint debug list item.
